sklearn/ensemble/tests/test_gradient_boosting.py

```diff
@@ -453,28 +453,27 @@ def test_max_feature_regression():
 
 
 def test_feature_importance_regression():
-    # Test to make sure Gini importance is calculated correctly
-    # Make sure that results are summed, then normalized to avoid
-    # over-weighting features from later stages
-    boston = datasets.load_boston()
-    X, y = boston.data, boston.target
-    X_train, X_test, y_train, y_test = train_test_split(X, y, shuffle=True,
-                                                        random_state=1)
-
-    clf = GradientBoostingRegressor(n_estimators=500, min_samples_split=2,
-                                    max_depth=4, learning_rate=0.01, loss='ls',
-                                    random_state=1)
-    clf.fit(X_train, y_train)
-
-    feature_importance = clf.feature_importances_
-    feature_importance = 100.0 * (feature_importance /
-                                  feature_importance.max())
-    sorted_idx = np.argsort(feature_importance)
-
-    ordered_features = ['ZN', 'CHAS', 'RAD', 'INDUS', 'B', 'TAX', 'AGE',
-                        'CRIM', 'NOX', 'PTRATIO', 'DIS', 'RM', 'LSTAT']
-
-    assert boston.feature_names[sorted_idx].tolist() == ordered_features
+    """Test that Gini importance is calculated correctly
+
+    This test is follows an example from [1] (pg. 373).
+
+    [1]: Friedman, J., Hastie, T., & Tibshirani, R. (2001). The elements of
+    statistical learning. New York: Springer series in statistics.
+    """
+    california = fetch_california_housing()
+    X, y = california.data, california.target
+    X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)
+
+    # Gradient boosting
+    reg = GradientBoostingRegressor(loss='huber', learning_rate=0.1,
+                                    max_leaf_nodes=6, n_estimators=200,
+                                    random_state=0)
+    reg.fit(X_train, y_train)
+    sorted_idx = np.argsort(reg.feature_importances_)[::-1]
+    feature_names_sorted = [california.feature_names[s] for s in sorted_idx]
+
+    assert feature_names_sorted[:4] == ['MedInc', 'Longitude',
```

Comment:
It might be worth testing the bottom 2-3 as well if they are stable. But at this point, I'm really happy with where this PR is.